lib/puppetserver/ca/config/puppet.rb

```diff
@@ -46,20 +46,33 @@ def initialize(supplied_config_path = nil)
         # on Windows are unsupported.
         # Note that Puppet Server runs as the [pe-]puppet user but to
         # start/stop it you must be root.
-        def user_specific_conf_dir
-          @user_specific_conf_dir ||=
-            if Puppetserver::Ca::Utils::Config.running_as_root?
+        def running_as_root?
+          @as_root ||= Puppetserver::Ca::Utils::Config.running_as_root?
+        end
+
+        def user_specific_puppet_confdir
+          @user_specific_puppet_confdir ||=
+            if running_as_root?
               '/etc/puppetlabs/puppet'
             else
               "#{ENV['HOME']}/.puppetlabs/etc/puppet"
             end
         end
 
-        def user_specific_conf_file
-          user_specific_conf_dir + '/puppet.conf'
+        def user_specific_puppetconf_file
+          user_specific_puppet_confdir + '/puppet.conf'
+        end
+
+        def user_specific_puppetserver_confdir
+          @user_specific_puppetserver_confdir ||=
+            if running_as_root?
+              '/etc/puppetlabs/puppetserver'
+            else
+              "#{ENV['HOME']}/.puppetserver"
+            end
         end
 
-        def load(cli_overrides = {})
+        def load(cli_overrides = {}, logger)
```

Comment:
This lets us log warnings without cause errors. Alternatively we could populate a `warnings` member that callers can handle like they currently do for `errors`.

Reply:
I think this is fine. To me, when an error happens, we want to do something else to handle it a lot of the time (like exiting 1). But warnings we just want to get logged and then we move on. So it makes sense to me to just do it inline.